vortex-array/src/compute/fill_null.rs

```diff
@@ -41,9 +41,9 @@ pub fn fill_null(array: impl AsRef<ArrayData>, fill_value: Scalar) -> VortexResu
         vortex_bail!(MismatchedTypes: array.dtype(), fill_value.dtype())
     }
 
-    array
-        .encoding()
-        .fill_null_fn()
-        .map(|f| f.fill_null(array, fill_value))
-        .unwrap_or_else(|| Err(vortex_err!(NotImplemented: "fill_null", array.encoding().id())))
+    if let Some(fill_null_fn) = array.encoding().fill_null_fn() {
+        return fill_null_fn.fill_null(array, fill_value);
+    }
+
+    fill_null(array.clone().into_canonical()?.into_array(), fill_value)
```

Comment:
You shouldn't fall back like this otherwise you will recurse if we forget to implement for a canonical array.

```
let array = array.clone().into_canonical()?.into_array();
if let Some(f) = array.encoding().fill_null_fn() {
   return ...
}
vortex_bail!("Missing canonical implementation of fill_null for dtype {}", array.dtype())
```